pkg/controller/dnszone/awsactuator.go

```diff
@@ -472,6 +472,107 @@ func (a *AWSActuator) Exists() (bool, error) {
 	return a.zoneID != nil, nil
 }
 
+func (a *AWSActuator) setInsufficientCredentialsConditionToFalse() bool {
+	accessDeniedConds, accessDeniedCondsChanged := controllerutils.SetDNSZoneConditionWithChangeCheck(
+		a.dnsZone.Status.Conditions,
+		hivev1.InsufficientCredentialsCondition,
+		corev1.ConditionFalse,
+		accessDeniedReason,
```

Comment:
This is not the correct reason.

Reply:
Are you sure?

That's not my code, I just refactored it here.

Reply:
I am sure that it is the wrong reason. I do see that it is an existing problem and not something introduce in this PR.

Reply:
What should the reason be then?

@akhil-rane git blame says that you added this. Do you have input on what this should be?

Reply:
In this case, the condition is being set to False. The reason should not be "AccessDenied" because access was not denied.